### PR TITLE
refactor!: Make `Node` opaque and optimize it for size

### PR DESCRIPTION
**Pull request by mwcampbell**
This is a massive breaking change for all users. But I'm afraid it's unavoidable.

This is still a work in progress. The technique is proven to work, but I still need to do the following:

* Use macros to eliminate as much getter/setter boilerplate as possible
* Get serialization, deserialization, and JSON schema support working again
* Use packed bit flags instead of individual `bool` fields

### CODE REVIEW

**Comment by mwcampbell** [comment]
Almost done. I need to implement the `JsonSchema` trait. Finally, because the base `Node` struct is now so small (currently 128 bytes), I think it no longer makes sense to box it (as we currently do using `Arc`).

**Comment by mwcampbell** [comment]
I was previously asked to implement the builder pattern, where the setter methods look like this:

```
pub fn with_role(mut self, role: Role) -> Self {
    self.role = role;
    self
}
```

I previously rejected this because I felt it was relying too much on the optimizer to avoid copying the struct around. I also felt that, while the pattern is convenient for hand-written test and demo code, it wouldn't matter much in actual GUI toolkit implementations, and might actually be less convenient in that context. Now I've realized that I was more right with that last point than I knew. In both egui and my Druid proof-of-concept integration, the toolkit stores under-construction `Node` structs in a collection, and makes mutable references to those structs available for the widget implementations to populate. A move-based builder pattern is incompatible with that implementation. So I'm now sure that my initial instinct was correct. Hand-rolled tests and samples will just have to be a bit more verbose.

**Comment by mwcampbell** [comment]
The change to eliminate the dependency on kurbo may seem to be over-reaching for this PR. But it's part of my plan to make AccessKit acceptable to even the most dependency-conscious projects, and I figured I should make all the planned breaking API changes in the foundational crate at once.

**Comment by mwcampbell** [comment]
This is now ready for review. The impact on a real GUI toolkit can be seen in [this egui branch](https://github.com/mwcampbell/egui/tree/accesskit-node-refactor).

@federicomenaquintero Since this is based on one of the techniques that you used to reduce memory usage in librsvg, as we previously discussed in the GNOME accessibility chat room, I'd appreciate your review.

File: common/src/lib.rs [hunk 1]
```diff
@@ -607,6 +607,46 @@ pub struct TextSelection {
     pub focus: TextPosition,
 }
 
+#[derive(EnumSetType, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "schemars", derive(JsonSchema))]
+#[cfg_attr(feature = "serde", serde(rename_all = "camelCase"))]
+#[cfg_attr(feature = "serde", enumset(serialize_as_list))]
+enum Flag {
```
[the remark this answers is not in the document]
Comment (federicomenaquintero) [reply common/src/lib.rs:701]: Ah, well, I see that you removed it later.  No problem, the point was compressing bools to bitflags.

**Comment by mwcampbell** [comment]
I have permission from @DataTriny (my primary collaborator on this project) to go ahead and merge this, though it hasn't yet been fully reviewed. We agree that the change is necessary, I'm confident that the approach is fundamentally sound, and I don't want to hold up corresponding updates in downstream projects. Please open issues or PRs for any problems found in subsequent reviews of this refactor.

**Pull request merged**